src/sema/mod.rs

```diff
@@ -614,6 +614,47 @@ impl ast::Namespace {
         None
     }
 
+    /// Resolve a contract name with namespace
+    pub fn resolve_contract_with_namespace(
```

Comment:
We shouldn't be adding all the functions in mod.rs, because it is getting too big. Perhaps we can create a file `namespace.rs` to contain the `impl` for `Namespace`. What do you think?

Reply:
I think it's a great idea. I've done this change now.

Now `src/sema/mod.rs` is nice and short.